tests/test_dataclasses.py

```diff
@@ -2944,3 +2944,30 @@ class Y:
         x: Union[X, None]
 
     assert Y(x={'y': None}).x.y is None
+
+
+@pytest.mark.parametrize('field_fn_unset', [Field, dataclasses.field])
+@pytest.mark.parametrize('field_fn2', [Field, dataclasses.field])
+def test_annotated_with_field_default_factory(field_fn_unset, field_fn2) -> None:
+    """
+    https://github.com/pydantic/pydantic/issues/9947
+    """
+
+    # wrap in lambda to create new instance for different fields
+    field1 = lambda: pydantic.Field(default_factory=lambda: 1)  # `Annotated` only works with pydantic field
+    field2 = lambda: field_fn2(default_factory=lambda: 2)
```

Comment:
Regarding ease of understanding, this pattern makes it a bit hard to digest for this tests. The cases that we're testing are simple enough and limited enough that I'd prefer we just explicitly test them all, ex: 

```py
a: Annotated[int, pydantic.Field(default_factory=lambda: 1)
```

It might be a bit more verbose at the end, but it's easier to understand quickly!